OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -62,21 +59,47 @@ public IEnumerable<string> ProvidesPrerequisites
 			}
 		}
 
+		protected override void Created(Actor self)
+		{
+			techTree = self.Owner.PlayerActor.Trait<TechTree>();
+			faction = self.Owner.Faction.InternalName;
```

Comment:
This needs to be done from the ctor so that it can use `init.Get<FactionInit, string>()` if it exists (see the original line).